core/src/main/java/org/elasticsearch/cluster/ClusterInfo.java

```diff
@@ -54,6 +54,11 @@ public Long getShardSize(ShardRouting shardRouting) {
         return shardSizes.get(shardIdentifierFromRouting(shardRouting));
     }
 
+    public long getShardSize(ShardRouting shardRouting, long defaultValue) {
```

Comment:
Instead of passing defaultValue here, can we just return ShardRouting.UNAVAILABLE_EXPECTED_SHARD_SIZE?  And maybe move that constant here?